Live Shuttle Bus Tracking

MomoG221 opened the pull request:
**Description**
1. Implemented functionality to display the position of different shuttle buses on the live map.
2. Updated the map view to fetch real-time data and accurately show shuttle bus locations.

**How was this tested?**
- [x] Manual Test: Compared shuttle bus positions on the map with the actual shuttle bus tracking on the Concordia website
Link to compare: https://shuttle.concordia.ca/concordiabusmap/
- [x] Unit Test

![IMG_4149](https://github.com/user-attachments/assets/e708557e-2463-4703-8f60-f900722ed9a5)
![IMG_4150](https://github.com/user-attachments/assets/1f4a3c8d-6753-4e80-9a9c-91ee7dc04310)
![IMG_4152](https://github.com/user-attachments/assets/fac7423f-ebb4-44ce-b158-fe90aab63a01)

shanvinluo commented:
LGTM, works on Android. I tested this at night, will retest it during the day tomorrow

MomoG221 commented:
@JeremyVieira Yes, it's supposed to be a callout box indicating the stop for the specific campus. I believe it might be the same bug you fixed earlier, as it's working fine for me on iOS. 
![IMG_4157](https://github.com/user-attachments/assets/c1dfecfe-aa90-454f-8cf0-9ac97416db59)
![IMG_4158](https://github.com/user-attachments/assets/626ef6a5-4584-402e-9db1-122774dabffd)


> Is there supposed to be something showing when you click the shuttle stop? ![image](https://private-user-images.githubusercontent.com/98359007/417667866-aa649825-d3f7-4bcd-8fdf-e6bac1456380.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA2NzEwNTAsIm5iZiI6MTc0MDY3MDc1MCwicGF0aCI6Ii85ODM1OTAwNy80MTc2Njc4NjYtYWE2NDk4MjUtZDNmNy00YmNkLThmZGYtZTZiYWMxNDU2MzgwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjI3VDE1MzkxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4MjcxZjc5NmE0Njk0NDNkYjNiYmQxMzJmODFhODk2YmQ4OTJjODRlM2U4N2ExNzdmZDY0ZTYwNzExMDA4ZjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ay20HbulULQrjyC6x-NGTsxSURiYR8xFKsVf5lNFWew)

JeremyVieira commented:
Removed the callout component from the ShuttleStop.js and now it renders on android. Still has to be checked on ios. Removed jest jsdom test environment as it wasn't being used and was causing a test suite to fail. Tests/application work on my end

MomoG221 commented:
Your modification didn’t make it work on iOS, so I reverted to what we had previously and used the CSS from the other marker we already implemented. It’s now working on iOS but still needs to be tested on Android.
![IMG_4159](https://github.com/user-attachments/assets/2baa500d-e694-4888-9eae-3c8f18d336d2)
![IMG_4160](https://github.com/user-attachments/assets/7c551ff4-ca28-4fd1-9501-d36c2d2ca39e)

MomoG221 commented:
I think I found why SonarQube was always showing 0% coverage. The problem was with the lcov report path; it wasn’t set correctly, so SonarQube couldn’t detect the test coverage. I updated it to ./coverage/lcov.info, which should allow SonarQube to properly pick up and report the coverage now.

JeremyVieira commented:
Everything working on my end, @MomoG221 confirmed that it is also working on ios
